src/Catel.Analyzers/Supressors/IDE0051OnPropertyChangeSupressor.cs

```diff
@@ -0,0 +1,159 @@
+        private static readonly string FodyAttributeLookupName = "Expose";
+
+        public override ImmutableArray<SuppressionDescriptor> SupportedSuppressions { get; } = ImmutableArray.Create(
+            new SuppressionDescriptor("CTLS0001", "IDE0051", "Supress IDE0051 on methods used for Fody INotifyPropertyChanged interception")
+        );
+
+        public override void ReportSuppressions(SuppressionAnalysisContext context)
+        {
+            try
+            {
+                // Check fody assembly
+                var fodyType = context.Compilation.GetTypeByMetadataName("PropertyChanged.AddINotifyPropertyChangedInterfaceAttribute");
+
+                if (fodyType is null)
+                {
+                    return;
+                }
+
+                // Get expose attribute instance
+                var fodyExposeType = context.Compilation.GetTypeByMetadataName("Catel.Fody.ExposeAttribute");
+
+                var supresssionDescriptor = SupportedSuppressions.FirstOrDefault();
+
+                foreach (var diagnostic in context.ReportedDiagnostics)
+                {
+                    var diagnosticSourceSyntaxTree = diagnostic.Location.SourceTree;
+
+                    if (diagnosticSourceSyntaxTree is null || !diagnostic.Location.IsInSource)
+                    {
+                        continue;
+                    }
+
+                    var rootSyntax = diagnosticSourceSyntaxTree.GetRoot(context.CancellationToken);
+
+                    var targetSyntax = rootSyntax.FindNode(diagnostic.Location.SourceSpan, false, true);
+
+                    // Look is class weaved by Fody
+                    var containerClassSyntax = targetSyntax.FirstAncestor<ClassDeclarationSyntax>();
+
+                    var rootSemantic = context.GetSemanticModel(diagnosticSourceSyntaxTree);
+                    var containerClassSymbol = (rootSemantic.GetDeclaredSymbol(containerClassSyntax, default) as ITypeSymbol);
+
+                    if (containerClassSymbol is null)
+                    {
+                        // Wrong class declaration
+                        continue;
+                    }
+
+                    if (!containerClassSymbol.InheritsFrom(FodyBaseClassLookupName))
+                    {
+                        continue;
+                    }
+
+                    var targetSymbol = rootSemantic.GetDeclaredSymbol(targetSyntax, context.CancellationToken);
+
+                    var accessModifierNode = targetSyntax.ChildTokens().FirstOrDefault(x => x.IsKind(Microsoft.CodeAnalysis.CSharp.SyntaxKind.PrivateKeyword));
+
+                    if (!accessModifierNode.IsKind(Microsoft.CodeAnalysis.CSharp.SyntaxKind.PrivateKeyword))
+                    {
+                        // Skip non-private
+                        continue;
+                    }
+
+                    if (targetSymbol is IMethodSymbol method && method.Name.StartsWith("On") && method.Name.EndsWith("Changed"))
```

Comment:
Try to use "continue" as much as possible to get out of this loop ASAP. So invert the if.